packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -1487,17 +1487,32 @@ mixin WidgetInspectorService {
     }
 
     if (config.includeProperties) {
-      List<DiagnosticsNode> properties = node.getProperties();
-      if (properties.isEmpty && value is Diagnosticable) {
```

Comment:
this was the problem case. The fix could have been just to remove these 4 lines which I added in error. They are not safe as cycles between property values and their properties could lead to a stack overflow. 

Inspecting this code I also noticed the previous code could sometimes override the json['properties'] field if a DiagnosticsProperty subclass returned properties as well has having a value that was Diagnosticable so I cleaned that case up.